flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/JobExceptionsHandler.java

```diff
@@ -119,7 +119,9 @@ private static JobExceptionsInfoWithHistory createJobExceptionsInfo(
         final ArchivedExecutionGraph executionGraph =
                 executionGraphInfo.getArchivedExecutionGraph();
         if (executionGraph.getFailureInfo() == null) {
-            return new JobExceptionsInfoWithHistory();
+            return new JobExceptionsInfoWithHistory(
```

Comment:
Is this really something that should be solved in the REST API? This looks more like an issue with the ArchivedExecutionGraph that the scheduler returns.

Reply:
Not 100% sure. The single root exception entry has been deprecated (at least from the API perspective). Need to check whether this failure info still makes sense.

https://github.com/apache/flink/blob/master/flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobExceptionsInfo.java#L45

Reply:
Anyway, this is a general problem of the execution graph with the AS, that we always return the current one (which might not have all the information) to the UI.

This is something we want to address eventually, but it should be a more systematic approach. Until that I'm fine with the current solution.

Reply:
To me this looks fine. The execution graph only contains the failureInfo if the job has been failed (which happens only once). This collides with restarting the job (which can happen multiple times).